sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -72,6 +73,8 @@ def __init__(
         parsed_url,  # type: Any
         service,  # type: str
         credential=None,  # type: Optional[Union[str, Dict[str, str], AzureNamedKeyCredential, AzureSasCredential, "TokenCredential"]] # pylint: disable=line-too-long
+        *,
+        audience: Optional[Union["BlobAudience", Any]],
```

Comment:
This wouldn't be Union `Any`, would probably be union with the other package's audiences, OR just `str` if we want to parse out the value as a `str` before it hits any `_shared` code.